Feature views

that-devguy opened the pull request:
Update to all styling, theme set to darkmode, updated forms, and dashboard (work in progress)
Includes Chelsea's fav buttons and dashboard routes

that-devguy committed:
updated classes for styling on login/newUser views

that-devguy committed:
updating dashboard layout and getting ready to pull main

that-devguy committed:
Merge branch 'main' of github.com:CodeByDex/PutASporkInIt into feature-views

ChelseaLuevano committed:
wrote the content for dashboard route to pull user recipes they created and recipes they favorited

ChelseaLuevano committed:
Ryan helped me write the helper.SafeRequest reference within the dashboard router function

that-devguy committed:
updated dashboard, updated login, newUser, and recipe-edit forms to be styled for darkmode

that-devguy committed:
Merge branch 'Feature-dashboard-routes' of github.com:CodeByDex/PutASporkInIt into feature-views

that-devguy committed:
updated dashboard nav route to include the userName by session.userID, updated dashboard layout, updated icon for logout in navbar

CodeByDex committed:
Merge branch 'main' of github.com:CodeByDex/PutASporkInIt into feature-views

CodeByDex commented:
Something on the render isn't working correctly, I see both sets of things opening but they aren't organizing like I think you were expecting them to.
![image](https://user-images.githubusercontent.com/119005046/236654117-ea55c6b3-2e19-45b6-ba48-19a9ceb429be.png)

that-devguy committed:
updated format on ingredients seed data, added a helper, updated form and added extra fields that needs to be linked up to models, updated form to change title to 'create new recipe' for add new and 'edit recipe' on edit

that-devguy committed:
Merge branch 'feature-views' of github.com:CodeByDex/PutASporkInIt into feature-views

that-devguy committed:
small changes to form

CodeByDex committed:
Merge branch 'main' of github.com:CodeByDex/PutASporkInIt into feature-views

CodeByDex committed:
fixed posting/put methods due to model updates

CodeByDex committed:
mapped client forms to post/put methods

CodeByDex committed:
Merge branch 'main' of github.com:CodeByDex/PutASporkInIt into feature-views

CodeByDex committed:
fixed merge issues with UOM drop down

CodeByDex reviewed (changes requested):
form needs updated for changes to recipe model before this PR can be completed.

that-devguy committed:
Merge branch 'main' of github.com:CodeByDex/PutASporkInIt into feature-views

that-devguy committed:
Merge branch 'feature-views' of github.com:CodeByDex/PutASporkInIt into feature-views

that-devguy committed:
added thumbnailURL to recipe model, updated create/edit form to include thumbnailURL, instructions, and ingredients, dashboard ui updates, logo & favicon update.

that-devguy committed:
updated model to use placeholder img as thumbnail when a user doesn't upload a url, updated recipe-preview and dashboard to fix recipe grid alignment

that-devguy commented:
Updated form - linked up new input fields with recipe model
Added thumbnailURL to recipe model - added logic to use a placeholder if the user's input is Null
Updated db/seeds to include thumbnailURL for all of the current recipes

CodeByDex committed:
Merge branch 'main' of github.com:CodeByDex/PutASporkInIt into feature-views

CodeByDex commented on controllers/navigationRoutes.js:
```diff
@@ -52,28 +52,53 @@ router.get("/recipe/:id/edit", async (req, res) => {
+        userID: req.session.userID
+      },
+    })
+
```
looks like we have some duplicate code issues with this function being in unsecured and secured.

CodeByDex commented on controllers/navigationRoutes.js:
```diff
@@ -157,8 +182,60 @@ router.get('/browse', async (req, res) => {
+      where: {
+        userID: req.session.userID
+      },
+   })
```
still need to put these through the renderRecipe function so that recipe preview renders correctly

CodeByDex commented on controllers/navigationRoutes.js:
```diff
@@ -157,8 +182,60 @@ router.get('/browse', async (req, res) => {
+  })
 })
 
+async function getRecipeViewModel(id) {
```
looks like another duplicate

CodeByDex commented on models/recipe.js:
```diff
@@ -68,9 +77,25 @@ Recipe.init(
       type: DataTypes.STRING,
       allowNull: false
     },
+    thumbnailURL: {
```
looks like thumbnail is listed twice.

CodeByDex commented on views/recipe-edit.handlebars:
```diff
@@ -1,75 +1,102 @@
+                                class="block mb-1 text-sm font-medium text-neutral-900 dark:text-white">Thumbnail URL</label>
+                            <input type="text" name="recipeThumbnail" id="recipeThumbnail"
+                                class="bg-neutral-50 border border-neutral-300 text-neutral-900 sm:text-sm rounded-lg focus:ring-green-500 focus:border-neutral-300 block w-full p-2.5 dark:bg-neutral-700 dark:border-neutral-600 dark:placeholder-neutral-400 dark:text-white caret-green-500 mb-2"
+                                placeholder="https://" value="{{thumbnail}}">
```
thinking this needs to be thumbnailURL, doesn't appear to be rendering correctly on the screen whatever the cause is. 
This recipe has a thumbnail but it didn't load on the form
![image](https://user-images.githubusercontent.com/119005046/236703924-f92b136c-3757-408c-ac49-b0b13c94df7e.png)

that-devguy committed:
fixed redundancies and fixed userRecipes and userRecipeFav

that-devguy committed:
merge from remote feature-views branch
Merge branch 'feature-views' of github.com:CodeByDex/PutASporkInIt into feature-views
